pkg/token/execCredentialWriter.go

```diff
@@ -20,20 +28,63 @@ type execCredentialWriter struct{}
 
 // Write writes the ExecCredential to standard output for kubectl.
 func (*execCredentialWriter) Write(token adal.Token) error {
-	t := v1.NewTime(token.Expires())
-	ec := &v1beta1.ExecCredential{
-		TypeMeta: v1.TypeMeta{
-			APIVersion: "client.authentication.k8s.io/v1beta1",
-			Kind:       "ExecCredential",
-		},
-		Status: &v1beta1.ExecCredentialStatus{
-			Token:               token.AccessToken,
-			ExpirationTimestamp: &t,
-		},
+	apiVersionFromEnv, err := helperGetApiVersionFromEnv()
+	if err != nil {
+		return err
 	}
+
+	var ec interface{}
+	t := metav1.NewTime(token.Expires())
+	switch apiVersionFromEnv {
+	case apiV1beta1:
+		ec = &v1beta1.ExecCredential{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: apiV1beta1,
+				Kind:       "ExecCredential",
+			},
+			Status: &v1beta1.ExecCredentialStatus{
+				Token:               token.AccessToken,
+				ExpirationTimestamp: &t,
+			},
+		}
+	case apiV1:
+		ec = &v1.ExecCredential{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: apiV1,
+				Kind:       "ExecCredential",
+			},
+			Status: &v1.ExecCredentialStatus{
+				Token:               token.AccessToken,
+				ExpirationTimestamp: &t,
+			},
+		}
+	}
+
 	e := json.NewEncoder(os.Stdout)
 	if err := e.Encode(ec); err != nil {
 		return fmt.Errorf("could not write the ExecCredential: %s", err)
 	}
 	return nil
 }
+
+func helperGetApiVersionFromEnv() (string, error) {
```

Comment:
nit: you don't need helper prefix. how about `getAPIVersionFromExecInfoEnv`

Reply:
Changed in new commit d64de9f. Please check it out. Thanks!